### PR TITLE
[qa] Replace with openwisp-utils-qa-checks

### PR DESCRIPTION
**Pull request by waleko**
Replace old qa checks with `openwisp-utils-qa-checks` (related to openwisp/openwisp-utils#63 for this repo)

### CODE REVIEW

**Comment by waleko** [comment]
Travis build is failing because of `flake8` fail ([travis](https://travis-ci.org/openwisp/ansible-openwisp2/jobs/631863747#L1296)). We can either ignore the whole `templates` folder or refactor the code later.

**Comment by atb00ker** [comment]
> Travis build is failing because of flake8 fail (travis). We can either ignore the whole templates folder or refactor the code later.

You can use autopep8 to fix these issues! :smile:

**Comment by waleko** [comment]
Although travis build succeeded, it behaves weirdly. I've rebuilt the same commit multiple times, and presumably randomly the tests failed. The qa part works flawlessly, but still we should fix it.
Here is the list of recent buids:
- https://travis-ci.org/openwisp/ansible-openwisp2/builds/633973065?utm_medium=notification&utm_source=github_status
- https://travis-ci.org/openwisp/ansible-openwisp2/builds/633970416?utm_medium=notification&utm_source=github_status
- https://travis-ci.org/openwisp/ansible-openwisp2/builds/633900948?utm_medium=notification&utm_source=github_status

**Comment by atb00ker** [comment]
> https://travis-ci.org/openwisp/ansible-openwisp2/builds/633900948?utm_medium=notification&utm_source=github_status

I think this fails because of missing endline. Am I suppose to focus on something here again? Did you not fix it and it runs automatically?

> https://travis-ci.org/openwisp/ansible-openwisp2/builds/633970416?utm_medium=notification&utm_source=github_status

Restarted this build and it failed again. Interesting. Have you done any investigation on why this is failing?

**Comment by waleko** [comment]
@atb00ker Sorry for the commit with missing endline, but you can still see, that `runtests.sh` [is failing](https://travis-ci.org/openwisp/ansible-openwisp2/jobs/633900953#L1581) on `Idempotence test`, although on other distors in the same build it works ok (and those failing distros aren't the same every build). You can see this behavior in the master branch as well:
- [c38b8fd](https://travis-ci.org/openwisp/ansible-openwisp2/builds/626669311?utm_source=github_status&utm_medium=notification)
- [abcff2c](https://travis-ci.org/openwisp/ansible-openwisp2/builds/626729788?utm_source=github_status&utm_medium=notification)
But in `f078a53` build and earlier travis builds [seemingly time out](https://travis-ci.org/openwisp/ansible-openwisp2/jobs/618790233#L418)? And in the next commit ([139b14](https://github.com/openwisp/ansible-openwisp2/commit/139b14594f314a02e6864f7c6e03672dc9732dde)) travis succeeds and in the commit after that([abcff2](https://github.com/openwisp/ansible-openwisp2/commit/abcff2c7fe3a22e391e5716f7a6b560e1481216d)) travis [fails](https://travis-ci.org/openwisp/ansible-openwisp2/jobs/626729795#L1231) the way we are familiar with. This is very bizarre.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[flake8]
+exclude = *migrations*,
+	  ./tests/*settings*.py,
+	  ./*setup*.py,
```
Comment (atb00ker) [inline setup.cfg]: @nemesisdesign Should we make setup.py flake8 compliant or ignore it?

**Comment by atb00ker** [comment]
>  that runtests.sh is failing on Idempotence test, although on other distors in the same build it works ok

Opened a new issue for it! :smile:

We can discuss it there! :smile:
Timeout happens sometimes and re-running travis will fix it! :smile:

File: setup.cfg (hunk 1 above)
Comment (nemesisdesign) [reply setup.cfg]: why do we need `flake8` and `isort` here? Do we have python code?
Maybe we do, but we don't have any `migrations` or `setup.py` I believe, therefore there's no need to ignore these files if I'm not missing anything.

File: .travis.yml [hunk 2]
```diff
@@ -44,15 +44,8 @@ script:
...
-  - pip install --user https://github.com/openwisp/openwisp-utils/tarball/master
-  - |
-    if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
-      # gets commit message of last commit before pull request merge
-      COMMIT_MESSAGE=$(git log $TRAVIS_PULL_REQUEST_SHA --format=%B -n 1)
-      printf "Checking commit message:\n\n"
-      printf "$COMMIT_MESSAGE\n\n"
-      checkcommit --message "$COMMIT_MESSAGE"
-    fi
+  - pip install --user "openwisp-utils[qa]>=0.3.0"
```
Comment (nemesisdesign) [inline .travis.yml]: use `0.3.1`

**Comment by waleko** [comment]
Removed `flake8` and `isort`

**Pull request merged**